src/storage/copier/csv_reader.cpp

```diff
@@ -0,0 +1,515 @@
+        auto value = TableCopyUtils::getVarListValue(
+            str_val, 1, str_val.length() - 2, type, copyDescription);
+        ValueVector::copyValueToVector(parse_chunk.getValueVector(column)->getData(),
+            parse_chunk.getValueVector(column).get(), value.get());
+        columnSizes[column]++;
+    } break;
+    case LogicalTypeID::FIXED_LIST: {
+        auto copyDescription =
+            CopyDescription({filePath}, common::CopyDescription::FileType::CSV, *csvReaderConfig);
+        auto value = TableCopyUtils::getArrowFixedListVal(
+            str_val, 1, str_val.length() - 2, type, copyDescription);
+        ValueVector::copyValueToVector(parse_chunk.getValueVector(column)->getData(),
+            parse_chunk.getValueVector(column).get() + columnSizes[column], value.get());
+        columnSizes[column]++;
+    } break;
+    default:
+        throw CopyException("Unsupported data type " +
+                            LogicalTypeUtils::dataTypeToString(type.getLogicalTypeID()) +
+                            " inside LIST");
+    }
+    parse_chunk.getValueVector(column)->state->selVector->selectedSize++;
+    // move to the next column
+    column++;
+}
+
+bool BaseCSVReader::AddRow(
+    DataChunk& insert_chunk, column_id_t& column, std::string& error_message, uint64_t buffer_idx) {
+    if (mode == ParserMode::PARSING_HEADER) {
+        return true;
+    }
+    linenr++;
+    if (row_empty) {
+        row_empty = false;
+        if (return_types.size() != 1) {
+            if (mode == ParserMode::PARSING) {
+                // Set This position to be null
+                ;
+            }
+            column = 0;
+            return false;
+        }
+    }
+    if (column < return_types.size()) {
+        throw CopyException(StringUtils::string_format(
+            "Error in file {} on line {}: expected {} values per row, but got {}", filePath, linenr,
+            return_types.size(), column));
+    }
+    if (mode == ParserMode::PARSING &&
+        columnSizes[0] >= common::StorageConstants::NODE_GROUP_SIZE) {
+        Flush(insert_chunk, buffer_idx);
+        return true;
+    }
+
+    column = 0;
+    return false;
+}
+
+bool BaseCSVReader::Flush(DataChunk& insert_chunk, uint64_t buffer_idx, bool try_add_line) {
+    if (parse_chunk.getNumValueVectors() == 0) {
+        return true;
+    }
+    for (uint64_t c = 0; c < parse_chunk.getNumValueVectors(); c++) {
+        auto parse_vector = parse_chunk.getValueVector(c);
+        auto result_vector = insert_chunk.getValueVector(c);
+        insert_chunk.getValueVector(c)->state->selVector->selectedSize =
+            parse_chunk.getValueVector(c)->state->selVector->selectedSize;
+        insert_chunk.insert(c, parse_chunk.getValueVector(c));
+    }
+    InitParseChunk(parse_chunk.getNumValueVectors());
+    return true;
+}
+
+void BaseCSVReader::InitParseChunk(column_id_t numCols) {
+    parse_chunk = DataChunk(numCols);
```

Comment:
This is not correct. You have to initialize internal value vectors and insert them to parse_chunk.

Also, if this is the correct place for init parse chunk, then you shouldn't init it at the first place in the constructor.